src/Servers/Kestrel/Core/src/Internal/Http/Http1ChunkedEncodingMessageBody.cs

```diff
@@ -59,6 +59,31 @@ public override bool TryReadInternal(out ReadResult readResult)
             return boolResult;
         }
 
+        public override async ValueTask<ReadResult> ReadAtLeastAsyncInternal(int minimumSize, CancellationToken cancellationToken = default)
+        {
+            await TryStartAsync();
+
+            try
+            {
+                var readAwaitable = _requestBodyPipe.Reader.ReadAtLeastAsync(minimumSize, cancellationToken);
+
+                _readResult = await StartTimingReadAsync(readAwaitable, cancellationToken);
```

Comment:
I'm worried that if `minimumSize` gets too large this could negatively affect our min request body data rate limit. We don't have logic like we do on the output which takes into account the size of the buffer. The logic assumes that if you're awaiting a read you'll get data as soon as it arrives.

Reply:
This is a good catch. Not yielding breaks that pattern here. This means we won't be able to take advantage of this feature (which is unfortunate)

Reply:
Or maybe we do what we do for writers and compute the read timeout based on the length?